### PR TITLE
Use nvcomp's snappy decompression in parquet reader

### PR DESCRIPTION
**Pull request by devavret**
Adds option to use nvcomp's batched snappy decompression functions in parquet writer.
By default, cuIO's internal snappy decompressor is used. Use export `LIBCUDF_USE_NVCOMP=1` to switch to nvcomp's decompressor.

### CODE REVIEW

**Review by vuule — COMMENTED** [review]
LGTM, once the writer changes are merged, this one is good to go as well IMO.

File: cpp/src/io/parquet/reader_impl.cu
Comment (hyperbolic2346) [inline cpp/src/io/parquet/reader_impl.cu:1021]: I know this isn't the case here, but I'm always nervous about a macro with a side-effect. I'm used to seeing these things fail because the assert macro was compiled out or changed. This isn't critical here, but it's easy to read over the initialization of `temp_size` inside a macro.

```suggestion
  auto const res = nvcompBatchedSnappyDecompressGetTempSize(
                                                  num_comp_pages, max_uncomp_page_size, &temp_size);
  CUDF_EXPECTS(res == nvcompStatus_t::nvcompSuccess,
               "Unable to get scratch size for snappy decompression");
```

File: cpp/src/io/parquet/reader_impl.cu
Comment (hyperbolic2346) [inline cpp/src/io/parquet/reader_impl.cu:1137]: This change to a struct makes the code much easier to follow and harder to get wrong. I like this.

**Review by hyperbolic2346 — COMMENTED** [review]
Overall this looks good to me, just a minor nit.

**Review by vuule — APPROVED** [review]

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by devavret** [comment]
rerun tests

**Comment by devavret** [comment]
@gpucibot merge

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
